### PR TITLE
[FINISHED] Refactors languages, adds missing languages and some species updates.

### PR DESCRIPTION
**Pull request by yorii**
## About The Pull Request

Updates mainly stuff from the following PRs:
https://github.com/tgstation/tgstation/pull/44336
https://github.com/tgstation/tgstation/pull/48106
https://github.com/tgstation/tgstation/pull/48309
~~https://github.com/tgstation/tgstation/pull/48573~~ Skipped this for now, but all previous additions to ethereals included.

But also a few other minor related additions and edits directly copied from tgstation, full changelog will be edited later.

## Why It's Good For The Game

Just brings some files that haven't been touched in a long while more up to date and includes the full assortment of languages, the tgui language menu, some language refactoring, borg charging station refactoring, and also some ethereal updates from tgstation. (all except the overcharge ability, can easily be added later if wanted)

Also adds some extra flavor to IPCs, giving them access to more languages than other races and they can also recharge from borg charging stations now.

## Changelog
:cl: yorii
add: added ethereal stomach and tongue
add: added all the species languages
add: added Uncommon language and foreigner neutral trait.
add: bone hurting juice hurts skeletons more
tweak: tweaked borg charger code to support charging non-borg mobs via signals (minor tg refactor)
code: adjusted a lot of language-related code formatting and a few other things to be in line with tg main branch.
balance: IPCs speak a subset of languages that are common among crew.
balance: IPCs can now charge their nutrition from borg charging stations
refactor: refactored language holder and all language related code
admin: added split for speaking/understanding languages in language menu (TGUI)
/:cl:

PS: Has a TGUI rebuild.

### CODE REVIEW

**Comment by yorii** [comment]
Plz review now

**Comment by yorii** [comment]
I mean now

**Comment by YakiAttaki** [comment]
If it helps, I already ported most of the racial language stuff to AuStation downstream, here:
https://github.com/austation/austation/pull/1712

**Comment by yorii** [comment]
> If it helps, I already ported most of the racial language stuff to AuStation downstream, here:
> [austation/austation#1712](https://github.com/austation/austation/pull/1712)

Thanks, I think I got all of it :) (and more!)

**Comment by yorii** [comment]
Ooops forgot to make the robots able to charge :)

**Comment by yorii** [comment]
TODO: make organics take a minor amount of burn damage if they're ever so stupid to try and connect themselves in a recharge station. And also something for IPCs.

**Comment by yorii** [comment]
Okay now I'm done :D I tested it too

**Comment by yorii** [comment]
I am now 100% done and I leave the rest up to whoever finds it.
Only thing really is if you want something to happen to non-electric races when they step into a charging station?
For now I leave it be as it is :D (that is nothing happens)

**Comment by yorii** [comment]
Hi I couldn't log in to github all day yesterday, but there are a few old proc calls that need to be adjusted with new source vars. (They'll still work fine with the source var of TRUE/FALSE though so this is actually mergeable as is).

I'll get to fixing the last few edits later this week.

**Comment by yorii** [comment]
it's finished and ready to merge now

**Comment by yorii** [comment]
oopsiewoopsie I made a wittle fuckywucky

**Comment by yorii** [comment]
there all good now :D

**Comment by yorii** [comment]
@zeskorion merge plz

**Review by qwertyquerty — DISMISSED** [review]

**Comment by yorii** [comment]
@ike709 you can testmerge this now, or whenever :) I'm looking forward to speaking secret languages with my racial brethren!

**Comment by yorii** [comment]
@zeskorion @ike709 come on please testmerge this before it becomes completely stale and too troublesome to unconflict...

**Comment by utilbird** [comment]
Hurry up and merge/close so we can fix shit downstream smh, it’s been almost a month

**Comment by Crossedfall** [comment]
I'll bug them about it and try for a test merge tomorrow. Not sure what the hold up has been on this one.

**Comment by Crossedfall** [comment]
Test merged onto Sage

File: code/modules/mob/living/carbon/human/species_types/mothmen.dm [hunk 1]
```diff
@@ -41,6 +42,13 @@
 	if(H.dna.features["moth_wings"] != "Burnt Off" && H.bodytemperature >= 800 && H.fire_stacks > 0) //do not go into the extremely hot light. you will not survive
 		to_chat(H, "<span class='danger'>Your precious wings burn to a crisp!</span>")
 		H.dna.features["moth_wings"] = "Burnt Off"
+		if(flying_species) //This is all exclusive to if the person has the effects of a potion of flight
```
Comment (zeskorion) [inline code/modules/mob/living/carbon/human/species_types/mothmen.dm]: can you remove all of these moth changes except the language one? i've handled them in gh#2143, in a far better way, and this'll just make conflicts

**Review by zeskorion — CHANGES_REQUESTED** [review]
other than requested changes, this looks pretty solid to me

**Comment by yorii** [comment]
Changes done :) Thanks for your review zesko!

**Comment by Crossedfall** [comment]
Updated TM

**Review by zeskorion — DISMISSED** [review]

**Comment by yorii** [comment]
What do with the dna.dm conflict?

**Comment by 4rchibald** [comment]
@yorii Plasmamen are no longer protected by their envirosuits and burst in flames the moment they spawn on arrival shuttle. Its clearly related to this PR since issue appeared during 2 rounds it was test merged and after quick searching I see some changes were made around the section of code related to suits and igniting when in oxygen atmosphere (Sadly I'm not a coder so I have no clue what exactly is the problem)

**Comment by yorii** [comment]
@4rchibald This PR doesn't really touch any of that.. is this happening on both Golden and Sage?

**Comment by yorii** [comment]
Actually nevermind, it does have some code cleanup regarding envirosuits, let me try to debug it!

**Comment by yorii** [comment]
I see what it is now when I tried to reproduce, plasmamen spawn in with their internals turned off for whatever reason..?
Turning on your internals as soon as the round starts prevents this from happening.

**Comment by yorii** [comment]
So yeah... I somehow made plasmemes spawn with their internals turned off instead of on. Let me see if I can rectify this.

**Comment by yorii** [comment]
Okay I fixed it now.

**Comment by yorii** [comment]
although somehow I managed to really spook the dna.dm file when resolving a merge conflict 👀

**Comment by yorii** [comment]
Can't unfuck dna.dm diff shenanigans, OCD triggered...

**Comment by 4rchibald** [comment]
@yorii What's currently test merged on Sage already includes your fix? 
Cause its still happening. Exactly same issue. Its not about internals (normally when they are off you are not burning, just suffocating) but enviro jumpsuits and/or helmets and how they protect plasmamen from bursting in flames when are in atmosphere with oxygen. The moment they join the game they are greeted with text 'Your body reacts with the atmosphere and bursts into flame!' and 'You're set on fire!' few moments later they are dead

**Comment by Crossedfall** [comment]
@4rchibald The TM does not include the latest changes. You can see the commit number in the MOTD. It was updated recently but the round hasn't ended yet. 

image

I have to temporarily pull this out to test another PR though, so the changes will be reflected when we test merge this again.

**Comment by yorii** [comment]
> @yorii What's currently test merged on Sage already includes your fix?
> Cause its still happening. Exactly same issue. Its not about internals (normally when they are off you are not burning, just suffocating) but enviro jumpsuits and/or helmets and how they protect plasmamen from bursting in flames when are in atmosphere with oxygen. The moment they join the game they are greeted with text 'Your body reacts with the atmosphere and bursts into flame!' and 'You're set on fire!' few moments later they are dead

Did you actually double check your internals? The only change which was missed was that internals at roundstart were off, and turning them on prevented fires from happening in my testing, it was confirmed by another person as well to be the roundstart internals setting.

In whichever case, the current version of the PR no longer causes plasmamen fire.

**Review by zeskorion — APPROVED** [review]

**Comment by Vivalas** [comment]
Yeah so this PR right here is why we test our code. This PR breaks ethereals. By changing the variable name in handle_charge from ethereal_charge to get_charge you broke ethereals, because there's no way to actually increment crystal_charge anywhere in the code now.

So, yay. :clap:

**Comment by yorii** [comment]
@Vivalas that wasn't the reason why it broke, the reason is that it's no longer the ethereal itself having the charge, it's their stomach organ. They can still charge with borg chargers and a fix for the rest is already on the way.

**Comment by Vivalas** [comment]
Yeah I was aware of that, crystal_charge is the variable stored in the ethereal stomach organ. I wasn't aware there was actually a way to charge them, however. I couldn't find anywhere in the code where crystal_charge was increased. interesting

**Comment by Crossedfall** [comment]
> 
> 
> Yeah I was aware of that, crystal_charge is the variable stored in the ethereal stomach organ. I wasn't aware there was actually a way to charge them, however. I couldn't find anywhere in the code where crystal_charge was increased. interesting

This is why we fully go through the code before making sarcastic rude comments.

So, yay. 👏

**Pull request merged**